Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +229,93 @@ namespace AzFramework
         }
     }
 
+    void XcbNativeWindow::DrawSplash()
+    {
+
+        int width = 0;
+        int height = 0;
+
+        auto get_image_data_static = [](int &width, int &height) -> uint8_t*
+        {
+            width = O3DE_LOGO::width;
+            height = O3DE_LOGO::height;
+
+            uint8_t *logo_binary_data = (uint8_t*)malloc(width * height * 4);   // so xcb_image_destroy can free it
+            
+            for (int i = 0; i < width * height; i++)
+            {
+                logo_binary_data[i * 4 + 0] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][0];
+                logo_binary_data[i * 4 + 1] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][1];
+                logo_binary_data[i * 4 + 2] = O3DE_LOGO::header_data_cmap[(unsigned char)O3DE_LOGO::header_data[i]][2];
+                logo_binary_data[i * 4 + 3] = 0xff;
+            }
+            return logo_binary_data;
+        };
+
+        uint8_t* data = get_image_data_static(width, height);
+
+        xcb_void_cookie_t cookie;
+        xcb_pixmap_t pixmap = xcb_generate_id(m_xcbConnection);
+        cookie = xcb_create_pixmap_checked(m_xcbConnection, m_xcbRootScreen->root_depth, pixmap, m_xcbWindow, width, height);
+        if (xcb_generic_error_t* error = xcb_request_check(m_xcbConnection, cookie))
```

Comment:
It looks like you are able to do this checkout without allocating data (line 255 above). It could save some resources (and in some cases) if it would be possible to move malloc after this if-statement and read only the necessary width and height.

Reply:
Yaa, but width and height are computed during loading of the image, I like that all image data is computed at the same time